spyder/plugins/preferences/widgets/configdialog.py

```diff
@@ -71,8 +72,9 @@ def __init__(self, parent=None):
         self.setLayout(vlayout)
 
         # Signals and slots
-        if self.main:
-            self.button_reset.clicked.connect(self.main.reset_spyder)
+        container = self.main.get_plugin(Plugins.Preferences).get_container()
+        self.button_reset.clicked.connect(
+            container.sig_reset_preferences_requested)
```

Comment:
This is not correct. The idea is for widgets to not have access to plugins. Instead, please add a signal to this dialog and emit it at this point like this:

```python
self.button_reset.clicked.connect(self.sig_reset_preferences_requested)
```

Then connect that signal to the container one of the same name when the dialog is created in this method:

https://github.com/spyder-ide/spyder/blob/552017c858cacc10a27285707c6e60651d0f49e7/spyder/plugins/preferences/widgets/container.py#L41-L42

like this:

```python
dlg.sig_reset_preferences_requested.connect(
    self.sig_reset_preferences_requested)
```

Reply:
something has to emit the `sig_reset_preferences_requested` instance from the preferences plugin container. What you're suggesting is another signal connection between the widget and the widget container; whatever the merits of this are, it will not result in resetting the preferences. Unless you meant that the widget's container `sig_reset_preferences_request` then be connected to the preferences plugin container?

Reply:
> Unless you meant that the widget's container sig_reset_preferences_request then be connected to the preferences plugin container's sig_reset_preferences_request?

Well, the container `sig_reset_preferences_request` is already connected to `self.reset`, and that's all that we need.

So, the thing is that Qt can chain signals without calling `emit` everywhere. So this code

    self.button_reset.clicked.connect(self.sig_reset_preferences_requested)

is equivalent to

    self.button_reset.clicked.connect(self.sig_reset_preferences_requested.emit)

and the same is true for this one:

    dlg.sig_reset_preferences_requested.connect(self.sig_reset_preferences_requested)

So when `button_reset` is clicked, it would emit `sig_reset_preferences_requested` at the dialog level, which then would emit `sig_reset_preferences_requested` at the container level, which finally would call `self.reset`.

I know this is convoluted, but (again) it's the standard for all other plugins. `ConfigDialog` is special because it has a reference to the main window, through which you can call `get_plugin`. But that's not the case for any other widget in the new API.

Reply:
Okay, I think I understand. Let me take another stab at it to see If I understand correctly.